core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -285,6 +287,20 @@ trait WorkflowComponent {
       groupedSeq.map(_.toMap)
     }
 
+    def countWorkflowsByQueueStatusByUser: ReadAction[Map[String, Map[String, Int]]] = {
+      // Query for a Seq of (User, Workflow) tuples.
+      val userWorkflows = (for {
+        workflow <- findQueuedAndRunningWorkflows
+        submission <- submissionQuery if workflow.submissionId === submission.id
+        user <- rawlsUserQuery if submission.submitterId === user.userSubjectId
+      } yield (user, workflow)).result
+
+      // Combine results into a nested map using Scalaz semigroups for aggregation.
```

Comment:
Jeff has been banging the "we can afford to FP a little harder" drum for a while and I think this might be the thing that pushes us over the edge. I'm running it by the other TLs now; watch this space, but my money's on:

1) general consensus that this is worth doing;
2) a preference for cats over scalaz; and
3) a tech talk or two to properly introduce a couple of these concepts to the team.